cpp/src/io/parquet/writer_impl.cu

```diff
@@ -531,7 +536,45 @@ std::vector<schema_tree_node> construct_schema_tree(
         }
       };
 
-      if (col->type().id() == type_id::STRUCT) {
+      auto last_list_child = [](cudf::detail::LinkedColPtr col) {
```

Comment:
IMO the name of a boolean should start with is_, since a boolean is returned.
Also, I don't understand the name. From the implementation and use it looks like it should be something like `can_output_lists_as_binary` ?

Reply:
I think `is_last_list_child` is appropriate here. It is asking if this is the last list before the leaf, so we can handle `list<list<list<int8>>>`. We want that last list. The output as binary check comes below before we ask about this level.

Reply:
:+1: to changing predicate functions to use an `is_` prefix.